lib/libzfs/libzfs_util.c

```diff
@@ -1940,3 +1942,66 @@ zprop_iter(zprop_func func, void *cb, boolean_t show_all, boolean_t ordered,
 {
 	return (zprop_iter_common(func, cb, show_all, ordered, type));
 }
+
+/*
+ * Fill given version buffer with zfs userland version
+ */
+void
+zfs_version_userland(char *version, int len)
+{
+	(void) strlcpy(version, ZFS_META_GITREV, len);
```

Comment:
@TerraTech I thought this was ready to go, but we're going to need to make one last change.  Using `ZFS_META_GITREV` here results in `zfs --version` printing `"unknown"` for the user land version when building from a zfs tarball (no git tree).  For example:

```sh
$ zfs version
unknown
zfs-kmod-0.8.0-rc3
```

That's clearly no good.  Can you change this back to the version and release which is what is used by the kmod.   This way, the user and kernel version strings will be reported consistently.  We can always investigate further extending the command to provide the `ZFS_META_GITREV` if we so choose.  Here's the local change I made to test this.

```diff
diff --git a/lib/libzfs/libzfs_util.c b/lib/libzfs/libzfs_util.c
index 7e097c1012..c0ca266c95 100644
--- a/lib/libzfs/libzfs_util.c
+++ b/lib/libzfs/libzfs_util.c
@@ -55,7 +55,6 @@
 #include <zfs_fletcher.h>
 #include <libzutil.h>
 #include <sys/zfs_sysfs.h>
-#include "zfs_gitrev.h"
 
 int
 libzfs_errno(libzfs_handle_t *hdl)
@@ -1949,7 +1948,8 @@ zprop_iter(zprop_func func, void *cb, boolean_t show_all, 
 void
 zfs_version_userland(char *version, int len)
 {
-       (void) strlcpy(version, ZFS_META_GITREV, len);
+       (void) strlcpy(version, "zfs-" ZFS_META_VERSION "-"
+           ZFS_META_RELEASE, len);
 }
 
 /*

```